### PR TITLE
7 add extrinsic to initiate redeem

### LINKED ISSUE #7: Add extrinsic to initiate redeem
"Redeem" = move tokens from Substrate to Stellar (which we called withdrawal in the past).

Add an extrinsic to the bridge pallet `initiate_redeem`. This extrinsic takes the following arguments:
* a wrapped asset (i.e. an asset managed an issued by the bridge pallet)
  * `asset_code: Vec<u8>`,
  * `asset_issuer: Vec<u8>`,
* `amount: BalanceOf<T>`,
* the Stellar account id of a vault account
  * also specified as a `Vec<u8>`

The extrinsic will
* check whether the user has a sufficient amount of these tokens
  * if not, it will return an error
* burn the specified amount of the wrapped asset

### PR DESCRIPTION
**Pull request by ashneverdawn**
Closes #7

### CODE REVIEW

File: pallets/spacewalk/src/lib.rs [hunk 1]
```diff
@@ -103,12 +103,26 @@ pub mod pallet {
             <T as frame_system::Config>::AccountId,
             BalanceOf<T>,
         ),
+        /// User initiated a redeem. [CurrencyIdOf<T>, T::AccountId, BalanceOf<T>]
```
Comment (ebma) [inline pallets/spacewalk/src/lib.rs:106]: This comment seems outdated

File: pallets/spacewalk/src/lib.rs [hunk 2]
```diff
@@ -131,14 +145,37 @@ pub mod pallet {
...
+ 
+        #[pallet::weight(100000)]
+        pub fn redeem(
+            origin: OriginFor<T>,
+            asset_code: Vec<u8>,
+            asset_issuer: Vec<u8>,
+            amount: BalanceOf<T>,
+            stellar_vault_pubkey: [u8; 32],
+        ) -> DispatchResultWithPostInfo {
+			let currency_id = T::StringCurrencyConversion::convert((asset_code.clone(), asset_issuer.clone()))
```
Comment (ebma) [inline pallets/spacewalk/src/lib.rs:162]: Indentation seems a bit off here

**Review by ebma — APPROVED** [review]
Looks good to me overall but I didn't test it.

File: .gitignore [hunk 3]
```diff
@@ -18,4 +18,7 @@
 
 .DS_Store
 .idea
-.vscode
```
Comment (gonzamontiel) [inline .gitignore:21]: Was this change intended?

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (gonzamontiel) [inline .vscode/launch.json:1]: I thought the `.vscode` folder should be ignored, is there a reason we want this for all the contributors?

**Review by gonzamontiel — CHANGES_REQUESTED** [review]

File: pallets/spacewalk/src/lib.rs (hunk 1 above)
Comment (ashneverdawn) [reply pallets/spacewalk/src/lib.rs]: fixed

File: pallets/spacewalk/src/lib.rs (hunk 2 above)
Comment (ashneverdawn) [reply pallets/spacewalk/src/lib.rs]: fixed

File: .gitignore (hunk 3 above)
Comment (ashneverdawn) [reply .gitignore:21]: no, resolved

**Review by gonzamontiel — APPROVED** [review]

**Pull request merged**